addons/account/wizard/account_reimbursement.py

```diff
@@ -0,0 +1,177 @@
+        self.payment_method_line_id = available_payment_method_line_ids[0]._origin if available_payment_method_line_ids else False
+
+    # -------------------------------------------------------------------------
+    # CONSTRAINT METHODS
+    # -------------------------------------------------------------------------
+    @api.constrains('amount', 'already_repaid_amount')
+    def _check_amount(self):
+        """ Ensures that the amount is <= payment to reimburse - already reimbursed amount"""
+        if not 0 < self.amount <= self.payment_id.amount - self.already_repaid_amount:
+            # TODO maybe store and indicate the maximum amount to pay
+            raise ValidationError(_('Reimbursement amount cannot exceed remaining amount to reimburse and must be superior to zero'))
+
+    # -------------------------------------------------------------------------
+    # LOW-LEVEL METHODS
+    # -------------------------------------------------------------------------
+    @api.model
+    def default_get(self, fields):
+        res = super(AccountReimbursement, self).default_get(fields)
+        payment_id = self.env['account.payment'].browse(self.env.context['active_id']) if self.env.context.get(
+            'active_model') == 'account.payment' else self.env['account.payment']
+        if payment_id.state != "posted":
+            raise UserError(_('You can only reimburse posted payments.'))
+        if 'company_id' in fields:
+            res['company_id'] = payment_id.company_id.id or self.env.company.id
+        if 'currency_id' in fields:
+            res['currency_id'] = payment_id.currency_id
+
+        return res
+
+    # -------------------------------------------------------------------------
+    # BUSINESS METHODS
+    # -------------------------------------------------------------------------
+    def reimburse_payments(self):
+        """ Reimburses a payment
+        First reverse moves related to the payment to reimburse and link it to the reversal moves
+        then create the reimbursement payment and link it to the reimbursed payment
+        then return an action window opening the newly created reimbursement payment
+        """
+        self.ensure_one()
+        payment_to_reimburse = self.payment_id
+
+        # create reimbursement payments
+        reimbursement_payment_id = self.env['account.payment'].create({
+            'date': self.reimbursement_date,
+            'amount': self.amount,
+            'payment_type': self.payment_type,
+            'partner_type': self.partner_type,
+            'partner_id': payment_to_reimburse.partner_id.id,
+            'ref': self.communication,
+            'currency_id': self.currency_id.id,
+            'payment_method_line_id': self.payment_method_line_id.id,
+            'journal_id': payment_to_reimburse.journal_id.id,
+            'company_id': self.company_id.id,
+            'reimbursed_payment_id': payment_to_reimburse.id,
```

Comment:
Please split function for facilitate inheritance in some custom module

`_prepare_reimburse_payment`

Reply:
Hello I will soon rework on this PR, your insight will be taken into account

Reply:
Do you think it's ready before V17 release?